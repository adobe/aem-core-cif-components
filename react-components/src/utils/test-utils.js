/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../__mocks__/i18nForTests';
import ConfigContextProvider from '../context/ConfigContext';
import UserContextProvider from '../context/UserContext';

import mutationChangePassword from './mocks/mutationChangePassword';
import mutationCreateCustomer from './mocks/mutationCreateCustomer';
import mutationCreateDuplicateCustomer from './mocks/mutationCreateCustomerDuplicate';
import mutationDeleteCustomerAddress from './mocks/mutationDeleteCustomerAddress';
import mutationGenerateToken from './mocks/mutationGenerateToken';
import mutationGenerateTokenWrongPassword from './mocks/mutationGenerateTokenWrongPassword';
import mutationMergeCarts from './mocks/mutationMergeCarts';
import mutationPlaceOrder from './mocks/mutationPlaceOrder';
import mutationRevokeToken from './mocks/mutationRevokeToken';
import mutationShippingAddress from './mocks/mutationShippingAddress';
import queryCart from './mocks/queryCart';
import queryCountries from './mocks/queryCountries';
import queryCustomerCart from './mocks/queryCustomerCart';
import queryCustomerDetails from './mocks/queryCustomerDetails';
import queryCustomerInformation from './mocks/queryCustomerInformation';
import queryEmptyCart from './mocks/queryEmptyCart';
import queryNewCart from './mocks/queryNewCart';

const defaultMocks = [
    mutationChangePassword,
    mutationCreateCustomer,
    mutationCreateDuplicateCustomer,
    mutationDeleteCustomerAddress,
    mutationGenerateToken,
    mutationGenerateTokenWrongPassword,
    mutationMergeCarts,
    mutationPlaceOrder,
    mutationRevokeToken,
    mutationShippingAddress,
    queryCart,
    queryCountries,
    queryCustomerCart,
    queryCustomerDetails,
    queryCustomerInformation,
    queryEmptyCart,
    queryNewCart
];

const defaultConfig = {
    storeView: 'default',
    graphqlEndpoint: 'none'
};

// eslint-disable-next-line react/display-name
const allProviders = (config, userContext, mocks) => ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            <ConfigContextProvider config={config || defaultConfig}>
                <UserContextProvider initialState={userContext}>
                    <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
                </UserContextProvider>
            </ConfigContextProvider>
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options = {}) => {
    const { config, userContext, mocks = defaultMocks, ...renderOptions } = options;
    return render(ui, { wrapper: allProviders(config, userContext, mocks), ...renderOptions });
};

export * from '@testing-library/react';
export { customRender as render };
