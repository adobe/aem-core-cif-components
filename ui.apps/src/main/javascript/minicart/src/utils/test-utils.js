/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import QUERY_CART_DETAILS from '../queries/query_cart_details.graphql';

const emptyCartId = 'empty';
const mockCartId = '123ABC';

const mocks = [
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: emptyCartId
            }
        },
        result: {
            data: {
                cart: {
                    email: null,
                    shipping_addresses: [],
                    prices: {
                        grand_total: {
                            currency: 'USD',
                            value: 0
                        }
                    },
                    selected_payment_method: {
                        code: '',
                        title: ''
                    },
                    billing_address: {
                        city: null,
                        country: {
                            code: null
                        },
                        lastname: null,
                        firstname: null,
                        region: {
                            code: null
                        },
                        street: [''],
                        postcode: null,
                        telephone: null
                    },
                    available_payment_methods: [
                        {
                            code: 'cashondelivery',
                            title: 'Cash On Delivery'
                        },
                        {
                            code: 'banktransfer',
                            title: 'Bank Transfer Payment'
                        },
                        {
                            code: 'checkmo',
                            title: 'Check / Money order'
                        },
                        {
                            code: 'free',
                            title: 'No Payment Information Required'
                        }
                    ],
                    items: []
                }
            }
        }
    },
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: mockCartId
            }
        },
        result: {
            data: {
                prices: {
                    grand_total: {
                        currency: 'USD',
                        value: 0
                    }
                },
                items: []
            }
        }
    }
];

const AllProviders = ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            {children}
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
