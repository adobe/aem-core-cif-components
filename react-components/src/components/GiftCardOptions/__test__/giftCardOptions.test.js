/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ~ Copyright 2021 Adobe
 ~
 ~ Licensed under the Apache License, Version 2.0 (the "License");
 ~ you may not use this file except in compliance with the License.
 ~ You may obtain a copy of the License at
 ~
 ~     http://www.apache.org/licenses/LICENSE-2.0
 ~
 ~ Unless required by applicable law or agreed to in writing, software
 ~ distributed under the License is distributed on an "AS IS" BASIS,
 ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ~ See the License for the specific language governing permissions and
 ~ limitations under the License.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
import React from 'react';
import { fireEvent, wait, screen } from '@testing-library/react';
import { render } from 'test-utils';
import GiftCardProductOptions from '../giftCardOptions';
import mockResponse from './graphqlMockGiftCardQuery';
import mockAddToCartMutation from './graphqlMockAddToCartMutation';

const config = {
    storeView: 'default',
    graphqlEndpoint: 'endpoint',
    graphqlMethod: 'GET',
    mountingPoints: {
        giftCardProductOptionsContainer: '#gift-card-product-options'
    }
};

describe('GiftCardProductOptions', () => {
    const eventHandler = jest.fn().mockImplementation(event => {
        return event.detail;
    });
    document.addEventListener('aem.cif.add-to-cart', eventHandler);
    document.addEventListener('aem.cif.add-to-wishlist', eventHandler);

    afterEach(() => {
        eventHandler.mockClear();
    });

    it('renders the component with no sku', () => {
        const { asFragment } = render(<GiftCardProductOptions />, { config: config });
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with sku', async () => {
        const gitfCardProductOptionsContainer = document.createElement('div');

        const { asFragment } = render(<GiftCardProductOptions sku="gift-card" />, {
            config: config,
            container: document.body.appendChild(gitfCardProductOptionsContainer),
            mocks: [mockResponse]
        });

        expect(await screen.findByText(/cart/i)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with wishlist', async () => {
        const gitfCardProductOptionsContainer = document.createElement('div');

        const { asFragment } = render(<GiftCardProductOptions sku="gift-card" showAddToWishList={true} />, {
            config: config,
            container: document.body.appendChild(gitfCardProductOptionsContainer),
            mocks: [mockResponse]
        });

        expect(await screen.findByText(/cart/i)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with full options', async () => {
        const gitfCardProductOptionsContainer = document.createElement('div');

        const { asFragment, getByRole, getByLabelText } = render(
            <GiftCardProductOptions sku="gift-card" showQuantity={true} />,
            {
                config: config,
                container: document.body.appendChild(gitfCardProductOptionsContainer),
                mocks: [mockResponse, mockAddToCartMutation]
            }
        );

        expect(await screen.findByText(/cart/i)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();

        // Click add to cart which should be disabled
        fireEvent.click(getByRole('button', { name: 'Add to Cart' }));
        expect(eventHandler).toHaveBeenCalledTimes(0);

        // Add the required inputs
        fireEvent.change(getByLabelText(/amount/i), {
            target: { value: 'Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEyLjAwMDA=' }
        });
        fireEvent.change(getByLabelText(/sender name/i), { target: { value: 'sender-name' } });
        fireEvent.change(getByLabelText(/sender email/i), { target: { value: 'sender-email' } });
        fireEvent.change(getByLabelText(/recipient name/i), { target: { value: 'recipient-name' } });
        fireEvent.change(getByLabelText(/recipient email/i), { target: { value: 'recipient-email' } });

        fireEvent.click(getByRole('button', { name: 'Add to Cart' }));

        // Add to cart should be called just once since the first click was on a disabled button
        await wait(() => expect(eventHandler).toHaveBeenCalledTimes(1));

        // The mock dispatchEvent function returns the CustomEvent detail
        expect(eventHandler).toHaveReturnedWith([
            {
                sku: 'gift-card',
                parentSku: 'gift-card',
                virtual: false,
                giftCard: true,
                quantity: 1,
                entered_options: [
                    {
                        uid: 'Z2lmdGNhcmQvZ2lmdGNhcmRfc2VuZGVyX25hbWU=',
                        value: 'sender-name'
                    },
                    {
                        uid: 'Z2lmdGNhcmQvZ2lmdGNhcmRfc2VuZGVyX2VtYWls',
                        value: 'sender-email'
                    },
                    {
                        uid: 'Z2lmdGNhcmQvZ2lmdGNhcmRfcmVjaXBpZW50X25hbWU=',
                        value: 'recipient-name'
                    },
                    {
                        uid: 'Z2lmdGNhcmQvZ2lmdGNhcmRfcmVjaXBpZW50X2VtYWls',
                        value: 'recipient-email'
                    },
                    {
                        uid: 'Z2lmdGNhcmQvZ2lmdGNhcmRfbWVzc2FnZQ==',
                        value: ''
                    }
                ],
                selected_options: ['Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEyLjAwMDA=']
            }
        ]);
    });

    it('renders add to wish list button', async () => {
        const gitfCardProductOptionsContainer = document.createElement('div');

        const { asFragment, getByRole } = render(<GiftCardProductOptions sku="gift-card" showAddToWishList={true} />, {
            config: config,
            container: document.body.appendChild(gitfCardProductOptionsContainer),
            mocks: [mockResponse, mockAddToCartMutation]
        });

        expect(await screen.findByText(/cart/i)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();

        // Click add to wish list button
        fireEvent.click(getByRole('button', { name: 'Add to Wish List' }));

        // Add to cart should be called just once since the first click was on a disabled button
        await wait(() => expect(eventHandler).toHaveBeenCalledTimes(1));

        expect(eventHandler).toHaveReturnedWith([
            {
                sku: 'gift-card',
                quantity: 1
            }
        ]);
    });
});
