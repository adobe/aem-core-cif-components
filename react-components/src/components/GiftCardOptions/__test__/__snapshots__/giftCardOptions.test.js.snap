// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftCardProductOptions renders the component with full options 1`] = `
<DocumentFragment>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Amount
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span
          class="fieldIcons__root"
        >
          <span
            class="fieldIcons__input"
          >
            <select
              aria-label="Amount"
              class="select__input field__input giftCardProduct__option"
            >
              <option
                value=""
              >
                Choose an Amount...
              </option>
              <option
                value="Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEyLjAwMDA="
              >
                12
              </option>
              <option
                value="Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEzLjAwMDA="
              >
                13
              </option>
              <option
                value="open_amount"
              >
                Other Amount...
              </option>
            </select>
          </span>
          <span
            class="fieldIcons__before"
          />
          <span
            class="fieldIcons__after"
          >
            <span
              class="icon__root"
            >
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Sender Name
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Sender Name"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Sender Email
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Sender Email"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Recipient Name
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Recipient Name"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Recipient Email
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Recipient Email"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Message
      </span>
       
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Message"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__quantity productFullDetail__section"
  >
    <h2
      class="productFullDetail__quantityTitle option__title"
    >
      <span>
        Quantity
      </span>
    </h2>
    <div
      class="quantity__root"
    >
      <span
        class="fieldIcons__root"
      >
        <span
          class="fieldIcons__input"
        >
          <select
            aria-label="Product quantity"
            class="select__input field__input"
            name="quantity"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
          </select>
        </span>
        <span
          class="fieldIcons__before"
        />
        <span
          class="fieldIcons__after"
        >
          <span
            class="icon__root"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="message-root"
      />
    </div>
  </section>
  <section
    class="productFullDetail__cartActions productFullDetail__section"
  >
    <button
      class="button__root_highPriority button__root clickable__root button__filled"
      disabled=""
      type="button"
    >
      <span
        class="button__content"
      >
        <span>
          Add to Cart
        </span>
      </span>
    </button>
  </section>
</DocumentFragment>
`;

exports[`GiftCardProductOptions renders the component with no sku 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <span
      class="indicator loader-img"
    />
    <span
      class="message"
    />
  </div>
</DocumentFragment>
`;

exports[`GiftCardProductOptions renders the component with sku 1`] = `
<DocumentFragment>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Amount
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span
          class="fieldIcons__root"
        >
          <span
            class="fieldIcons__input"
          >
            <select
              aria-label="Amount"
              class="select__input field__input giftCardProduct__option"
            >
              <option
                value=""
              >
                Choose an Amount...
              </option>
              <option
                value="Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEyLjAwMDA="
              >
                12
              </option>
              <option
                value="Z2lmdGNhcmQvZ2lmdGNhcmRfYW1vdW50LzEzLjAwMDA="
              >
                13
              </option>
              <option
                value="open_amount"
              >
                Other Amount...
              </option>
            </select>
          </span>
          <span
            class="fieldIcons__before"
          />
          <span
            class="fieldIcons__after"
          >
            <span
              class="icon__root"
            >
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Sender Name
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Sender Name"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Sender Email
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Sender Email"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Recipient Name
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Recipient Name"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Recipient Email
      </span>
       
      <span
        class="required"
      >
         *
      </span>
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Recipient Email"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__section productFullDetail__giftCardProduct"
  >
    <h3
      class="option__title"
    >
      <span>
        Message
      </span>
       
    </h3>
    <div>
      <div
        class="giftCardOptionSelect__root"
      >
        <span>
          <input
            aria-label="Message"
            class="field__input"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </section>
  <section
    class="productFullDetail__quantity productFullDetail__section"
  >
    <h2
      class="productFullDetail__quantityTitle option__title"
    >
      <span>
        Quantity
      </span>
    </h2>
    <div
      class="quantity__root"
    >
      <span
        class="fieldIcons__root"
      >
        <span
          class="fieldIcons__input"
        >
          <select
            aria-label="Product quantity"
            class="select__input field__input"
            name="quantity"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
          </select>
        </span>
        <span
          class="fieldIcons__before"
        />
        <span
          class="fieldIcons__after"
        >
          <span
            class="icon__root"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="message-root"
      />
    </div>
  </section>
  <section
    class="productFullDetail__cartActions productFullDetail__section"
  >
    <button
      class="button__root_highPriority button__root clickable__root button__filled"
      disabled=""
      type="button"
    >
      <span
        class="button__content"
      >
        <span>
          Add to Cart
        </span>
      </span>
    </button>
  </section>
</DocumentFragment>
`;
